web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx

```diff
@@ -203,36 +210,53 @@ const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
                             min={1}
                             max={selectedSellOrder?.quantity}
+                            validate={getCreditCountValidation(
+                              Number(selectedSellOrder?.quantity),
+                            )}
                           />
                         </div>
-                        <Title variant="h6" sx={{ mr: 4 }}>
-                          =
-                        </Title>
-                        <div
-                          className={cx(styles.flexColumn, styles.marginRight)}
+                        <Box
+                          sx={{
+                            display: 'flex',
+                            alignItems: 'center',
+                            height: 60,
+                          }}
                         >
-                          <Box
-                            sx={{
-                              display: 'flex',
-                              alignItems: 'baseline',
-                              flexWrap: 'wrap',
-                            }}
+                          <Title variant="h6" sx={{ mr: 4 }}>
+                            =
+                          </Title>
+                          <div
+                            className={cx(
+                              styles.flexColumn,
+                              styles.marginRight,
+                            )}
                           >
                             <Box
-                              sx={{ display: 'flex', alignItems: 'baseline' }}
+                              sx={{
+                                display: 'flex',
+                                alignItems: 'baseline',
+                                flexWrap: 'wrap',
+                              }}
                             >
-                              <RegenTokenIcon className={styles.regenIcon} />
-                              <Title variant="h4" sx={{ mr: 1.5 }}>
-                                {values.creditCount *
-                                  microToDenom(selectedSellOrder?.askAmount) ||
-                                  '-'}
-                              </Title>
+                              <Box
+                                sx={{ display: 'flex', alignItems: 'baseline' }}
+                              >
+                                <RegenTokenIcon className={styles.regenIcon} />
+                                <Title variant="h4" sx={{ mr: 1.5 }}>
+                                  {amountToSpend({
+                                    askAmount: Number(
+                                      selectedSellOrder?.askAmount,
+                                    ),
+                                    creditCount: values?.creditCount,
+                                  })}
+                                </Title>
+                              </Box>
+                              <Label size="sm" sx={{ color: 'info.dark' }}>
```

Comment:
just FYI - you can pass `color` directly if you want (`<Label size=“sm” color=“info.dark”>`